router/router.go

```diff
@@ -39,10 +39,21 @@ func NewGinEngine(ginMode string, version string, urlExpiration time.Duration, s
 
 		bucket := c.Param("bucket")
 		key := c.Param("key")
+		expiration := c.Param("expiration")
+
+		if expiration != "" {
```

Comment:
Comme tu l'as proposé hier, y'a moyen de factoriser un peu le code.

en vrai, sortir une fonction pour gérer ça peut sembler overkill (on va factoriser 3 lignes de code), mais l'interêt majeur sera de respecter un principe du Go idiomatique: le happy path, cf. https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

dans ce cas précis:
Dans Go, on préfère:

```go
if queryParam=="" {
  return defaultExpiration
}

res, err := time.ParseDuration(queryParam)
if err!=nil {
  // 400 error
}
 ```

à:

```go
if queryParam != "" {
  es, err := time.ParseDuration(queryParam)
  if err!=nil {
    // 400 error
  }
}
```

et le fait d'extraire une fonction ici te permet justement d'implémenter le happy path (car tu pourra utiliser des `return`:

```go
func parseExpiration(s string, fallback time.Duration) (time.Duration, error) {
	if s == "" {
		return fallback, nil
	}

	return time.ParseDuration(s)
}
```

ensuite à l'utilisation ça donnerait:

```go
presignedURLApiV1.POST("/:bucket/*key", func(c *gin.Context) {

		var (
			bucket     = c.Param("bucket")
			key        = c.Param("key")
			expiration = c.Param("expiration")
		)

		urlExpiration, err := parseExpiration(expiration, urlExpiration)
		if err != nil {
			c.JSON(http.StatusBadRequest, gin.H{"error": "Failed to parse Duration " + expiration})
			return
		}

		url, err := s3Backend.CreatePresignedURLForUpload(backend.BucketObject{BucketName: bucket, Key: key}, urlExpiration)
		if err != nil {
			log.Error("Failed to create presigned PutObject URL for %s %s", key, bucket, err)
			c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to create PutObject URL for " + key})
			return
		}

		c.JSON(http.StatusOK, gin.H{"url": url})
	})
```